tests/test_coordinator.py

```diff
@@ -161,16 +161,22 @@ def test_local_sensors():
         pass
 
     sensor_values = coordinator.get_sensors()
-    assert isinstance(sensor_values, dict)
-    for k, v in sensor_values.items():
-        assert isinstance(k, ValueName)
-        assert isinstance(v, SensorValueNew)
+    assert isinstance(sensor_values, SensorValues)
 
+    values_dict = sensor_values.to_dict()
+    # if all of the values are none, then this is just a blank object
+    assert(not all([v is None for v in values_dict.values()]))
+    for k, v in values_dict.items():
+        assert isinstance(k, ValueName)
+        assert isinstance(v, int) or isinstance(v, float) or v is None
 
+@pytest.mark.timeout(10)
 def test_remote_sensors():
     # wait before
-    while not is_port_in_use(rpc.default_port):
-        time.sleep(1)
+    # this fails every time and i'm not sure how it wouldnt? - jls
```

Comment:
I don't know how this is supposed to work but it failed every time, there seems to be some sort of serial dependence between the tests because I would get an error that the port was already bound when i did a keyboard escape. if there is some network connection that needs to be set up for all tests, you should read a bit about how pytest fixtures work -- i'm new to pytest but i definitely saw that you can set fixtures up to preserve state across tests, modules, functions, etc.

Reply:
I also think uncontrolled while loops are a bad pattern in any case in a critical-use program like this.